### ISSUE TITLE
Graceful error message when server fails to start with panic

### ISSUE THREAD

**Issue by ngjaying** [lf-edge/ekuiper#910]
For example, if required folder is missing, the error message should provide enough detail.

**Comment by wfnuser** [comment]
I would like to have a try. 😊

**Comment by ngjaying** [comment]
Great!

**Comment by wfnuser** [comment]
> ired folder is missing, the error message should provide enough detail.

The error message is something like this? `time="2021-08-17 18:02:14" level=fatal msg="conf dir not found, please set KuiperBaseKey program environment variable correctly." file="conf/conf.go:114"`

I'm trying to start to develop the ekuiper code. Frankly to say, the documents, especially the development guide, are lacking.

**Comment by ngjaying** [comment]
@wfnuser Yes, especially when other folder is missing, we still get the conf dir not found problem.

Regarding the development guide, we now only have the contributing.md, [debug your code](https://github.com/lf-edge/ekuiper/blob/master/docs/CONTRIBUTING.md#debug-your-code). We need to enrich that for sure. Any suggestions are welcomed.
